### DISCUSSION TITLE
Vertical Align for ui.label()

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1584] [Q&A] [answered]
I'm not sure if it's a bug ,the tailwindCSS  Vertical Align for ui.label()  doesn't work.      
```py
ui.label('test').classes('rounded bg-green-500  text-center h-20  w-40 inline-block align-middle text-center ')
```
![微信截图_20230908181224](https://github.com/zauberzeug/nicegui/assets/25096045/b4e1acdf-d5aa-4a20-9192-050a7f36d352)

**Comment by Anindya088** [comment, marked as answer]
Hi @v479038280 ,

Yes, I have also faced this 😄 Usually I will just nest it under a row and use the row Tailwind classes, somehow that works.
I don't really understand CSS, so don't know the difference between the 2, and why one works and the other does not. Just that realized that this could be a workaround.

```python
with ui.row().classes('rounded bg-green-500 h-20 w-40 place-content-center'):
    ui.label('test')
``` 

Thanks,
Anindya

**Reply by falkoschindler** [reply]
I aggree: Wrapping the text in a container is the way to go, see [this thread on StackOverflow](https://stackoverflow.com/questions/55056513/vertical-align-with-tailwind-css-across-full-screen-div).
`align-middle` does not work on the label, because it aligns the label _within its container_, not the text within the label.
